packages/cozy-client/src/models/file.js

```diff
@@ -30,3 +30,27 @@ export const isNote = file => {
     return true
   return false
 }
+
+/**
+ * Normalize an object representing a io.cozy.file object
+ *
+ * Ensure existence of `_id` and `_type` and define a `path`
+ * on files if provided the parent directory.
+ *
+ * @param {object} file - object representing the file
+ * @param {object} parent - object representing the parent directory
+ * @returns {object} full normalized object
+ */
+export function normalize(file, parent) {
+  const id = file._id || file.id
+  const type = file.type || file._type || 'io.cozy.files'
+  const path = parent && parent.path ? parent.path + '/' + file.name : undefined
+  return {
+    _id: id,
+    id: id,
+    _type: type,
+    type: type,
+    path: path,
+    ...file
+  }
```

Comment:
In other parts of cozy-client's code, normalize only deals with _id and _type. Here we add logic about the path, are we sure that it should be done in another function ?

Also, what happens if file has a path and parent has a path ? Here the file's path would override the parent's path, is is something that we want ? It seems like an unwanted side-effect, this is why I am a bit hesitant with having the path logic inside the normalize method.

Reply:
Logic was "Ensure it has a path, build it from parent if needed", like we ensure an _id and a _type (but won't override the document's ones if they are present)